### ISSUE TITLE
2019-12-20 Computation engine error for all M4SS and some M4SD testcases introduced in 5a9f91a

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#128]
### Error

When running M4SS testcases (tc008, tc018, tc105) the following error is given when using nominal time. This behaviour was introduced in 5a9f91aa0d. Example error below was provided using tc105, but similar messages are provide for all 3 M4SS tests.

```r
Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, paste0("AE.", sprintf("%.2f", unique(data_data[, : replacement has 2 rows, data has 1
For SDEID 661083556 
7. stop(paste0(e, "For SDEID ", unique(data_data[, map_data$SDEID])[i])) at run_M4_SS_computation.R#1615
6. value[[3L]](cond) 
5. tryCatchOne(expr, names, parentenv, handlers[[1L]]) 
4. tryCatchList(expr, classes, parentenv, handlers) 
3. tryCatch({
    if (comp_required[["DIi"]]) {
        di <- list()
    } ... at run_M4_SS_computation.R#698
2. run_M4_SS_computation(data = merged_data, map = map_data, method = method, 
    model_regex = model_regex, parameter_list = parameter_list, 
    return_list = return_list) at run_computation.R#2429
1. run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) 
```

**Comment by jhhughes256** [comment]
### Update

This error also occurs for tc008 and tc116 M4SD.

```r
 Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, paste0("AE.", sprintf("%.2f", unique(data_data[, : replacement has 2 rows, data has 1
For SDEID 225930823 
7. stop(paste0(e, "For SDEID ", unique(data_data[, map_data$SDEID])[i])) at run_M4_SD_computation.R#1608
6. value[[3L]](cond) 
5. tryCatchOne(expr, names, parentenv, handlers[[1L]]) 
4. tryCatchList(expr, classes, parentenv, handlers) 
3. tryCatch({
    tmp_df <- data_data[data_data[, map_data$SDEID] == unique(data_data[, 
        map_data$SDEID])[i], ]
    tmp_df <- tmp_df[order(tmp_df[, map_data$TIME]), ] ... at run_M4_SD_computation.R#751
2. run_M4_SD_computation(data = merged_data, map = map_data, method = method, 
    model_regex = model_regex, parameter_list = parameter_list, 
    return_list = return_list) at run_computation.R#2423
1. run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) 
```

**Comment by jhhughes256** [comment]
### Update

This error also occurs for tc111 M4SD specifically when using actual time. While slightly different to above, this does not occur in and was introduced in .

```r
 Error in value[[3L]](cond) : 
  Error in `[<-.data.frame`(`*tmp*`, i, paste0("AMT.", sprintf("%.2f", unique(data_data[, : replacement has 7 rows, data has 1
For SDEID 469690710 
7. stop(paste0(e, "For SDEID ", unique(data_data[, map_data$SDEID])[i])) at run_M4_SD_computation.R#1608
6. value[[3L]](cond) 
5. tryCatchOne(expr, names, parentenv, handlers[[1L]]) 
4. tryCatchList(expr, classes, parentenv, handlers) 
3. tryCatch({
    tmp_df <- data_data[data_data[, map_data$SDEID] == unique(data_data[, 
        map_data$SDEID])[i], ]
    tmp_df <- tmp_df[order(tmp_df[, map_data$TIME]), ] ... at run_M4_SD_computation.R#751
2. run_M4_SD_computation(data = merged_data, map = map_data, method = method, 
    model_regex = model_regex, parameter_list = parameter_list, 
    return_list = return_list) at run_computation.R#2423
1. run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) 
```

**Comment by jhhughes256** [comment]
This issue has been addressed through the work made in #64.
